internal/workflows/workflows.go

```diff
@@ -0,0 +1,168 @@
+package workflows
+
+import (
+	"fmt"
+
+	argo "github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1"
+	apiv1 "k8s.io/api/core/v1"
+	v1resources "k8s.io/apimachinery/pkg/api/resource"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"teainspace.com/ame/api/v1alpha1"
+)
+
+// TODO: Write tasks for the following:
+// TODO: Determine how to handle support for readwritemany vs readwriteonce
+// TODO: Determine what to do if the volume is not large enough
+// TODO: Secure the volume so only the pod from the workflow and read and write to it.
+// TODO: Optimize the steps so we don't download and upload project files, dependencies and artifacts multiple times.
+// TODO: Get logs working with pipelines
+
+func GenPipelineWf(t *v1alpha1.Task) (*argo.WorkflowSpec, error) {
+	var wfSteps []argo.ParallelSteps
+	for _, s := range t.Spec.Pipeline {
+		tSpec := genTaskSpecFromPipelineStep(t, s)
+		wfTemplate := genTemplate(s.TaskName, v1alpha1.NewTaskFromSpec(tSpec, t.GetName()+s.TaskName), taskVolName(t))
+		wfSteps = append(wfSteps, argo.ParallelSteps{
+			Steps: []argo.WorkflowStep{
+				{
+					Inline: &wfTemplate,
+					Name:   s.TaskName,
+				},
+			},
+		})
+	}
+
+	volStoreageReq, err := v1resources.ParseQuantity("5Gi")
+	if err != nil {
+		return nil, err
+	}
+
+	return &argo.WorkflowSpec{
+		PodMetadata: &argo.Metadata{
+			Labels: map[string]string{
+				"ame-task": t.GetName(),
+			},
+		},
+		Templates: []argo.Template{
+			{
+				Steps: wfSteps,
+				Name:  "main",
+			},
+		},
+		Entrypoint: "main",
+		VolumeClaimTemplates: []apiv1.PersistentVolumeClaim{
+			{
+				ObjectMeta: v1.ObjectMeta{
+					Name: taskVolName(t),
+				},
+				Spec: apiv1.PersistentVolumeClaimSpec{
+					AccessModes: []apiv1.PersistentVolumeAccessMode{
+						apiv1.ReadWriteOnce,
+					},
+					Resources: apiv1.ResourceRequirements{
+						Requests: apiv1.ResourceList{
+							"storage": volStoreageReq,
+						},
+					},
+				},
+			},
+		},
+	}, nil
+}
+
+func taskVolName(t *v1alpha1.Task) string {
+	return fmt.Sprintf("%s-volume", t.GetName())
+}
+
+func genTaskSpecFromPipelineStep(parentTask *v1alpha1.Task, step v1alpha1.PipelineStep) *v1alpha1.TaskSpec {
+	spec := parentTask.Spec.DeepCopy()
+	spec.RunCommand = step.RunCommand
+	spec.Pipeline = nil
+	spec.Env = step.Env
+	spec.Secrets = step.Secrets
+	return spec
+}
+
+func genTemplate(templateName string, t *v1alpha1.Task, volName string) argo.Template {
```

Comment:
I think it needs to be more clear what kind of template this function generates.